src/Rules/DoctrineKeyValueStyleRule.php

```diff
@@ -164,12 +168,27 @@ public function processNode(Node $callLike, Scope $scope): array
                     continue;
                 }
 
-                // Be a bit generous here by allowing column value types that
-                // are *maybe* accepted. If we want to be strict, we can warn
-                // unless they are *definitely* accepted.
+                $argColumnType = $argColumn->getType();
                 $valueType = $argType->getValueTypes()[$keyIndex];
-                if ($argColumn->getType()->accepts($valueType, true)->no()) {
-                    $errors[] = 'Column "' . $table->getName() . '.' . $argColumnName . '" expects value type ' . $argColumn->getType()->describe(VerbosityLevel::precise()) . ', got type ' . $valueType->describe(VerbosityLevel::precise());
+
+                // Convert IntegerRangeType column types into IntegerType so
+                // that any integer value is accepted for integer columns,
+                // since it is uncommon to check integer value ranges.
+                if ($argColumnType instanceof IntegerRangeType) {
+                    $argColumnType = new IntegerType();
+                } elseif ($argColumnType instanceof UnionType) {
+                    $newTypes = [];
+                    foreach ($argColumnType->getTypes() as $type) {
+                        if ($type instanceof IntegerRangeType) {
+                            $type = new IntegerType();
+                        }
+                        $newTypes[] = $type;
+                    }
+                    $argColumnType = TypeCombinator::union(...$newTypes);
+                }
+
+                if (! $argColumnType->accepts($valueType, true)->yes()) {
```

Comment:
I guess the cause of all our problems is that we call `accepts` here instead of `isSuperTypeOf`

```suggestion
                if (! $argColumnType->isSuperTypeOf($valueType)->yes()) {
```

Reply:
after we use isSuperTypeOf, I think we can drop the IntegerRange handling above